From 9373456f7b3f3e34e048d1af0d0d336d814526e8 Mon Sep 17 00:00:00 2001
From: William Horvath <william@horvath.blog>
Date: Fri, 22 Nov 2024 10:31:07 -0800
Subject: [PATCH] configure: Enforce 4-byte stack alignment on clang for PE
 builds.

Same as -mpreferred-stack-boundary=2 for gcc.
Clang interprets this as "n" bytes, while gcc uses "2^n" bytes.
https://linux.kernel.narkive.com/y7hAplfS/patch-x86-build-specify-stack-alignment-for-clang
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index b63f3c391c9..0e7bdc99eb3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -620,6 +620,7 @@ Please install mingw-w64, clang/llvm-dlltool/lld, or llvm-mingw.])
 
     case $wine_arch in
       i386) WINE_TRY_PE_CFLAGS([-fno-omit-frame-pointer])
+            WINE_TRY_PE_CFLAGS([-mstack-alignment=4])
             WINE_TRY_PE_CFLAGS([-mpreferred-stack-boundary=2])
             WINE_TRY_PE_CFLAGS([-mlong-double-64])
             WINE_TRY_PE_CFLAGS([-msse2])
-- 
2.56.0

